docs: clarify package name rules (#1326)
npm/documentation: content/packages-and-modules/contributing-packages-to-the-registry/creating-a-package-json-file.mdx

A `package.json` file must contain `"name"` and `"version"` fields.

> The `"name"` field contains your package's name, and must be lowercase and one word, and may contain hyphens, dots and underscores.

Reviewer: ```suggestion
The `"name"` field contains your package's name, and must be lowercase, not contain any spaces, and may contain hyphens, dots, and underscores.
```

"one word" seemed confusing, cause `my-cool-package` is definitely a valid package name.  What we mean is no spaces.

Also, the Oxford comma is very near and dear to me so let's include it.
